bson/src/main/scala/org/mongodb/scala/bson/codecs/Macros.scala

```diff
@@ -83,4 +84,23 @@ object Macros {
   @compileTimeOnly("Creating a Codec utilises Macros and must be run at compile time.")
   def createCodec[T](codecRegistry: CodecRegistry): Codec[T] = macro CaseClassCodec.createCodec[T]
 
+  /**
+   * Annotations to use on case classes that are being processed by macros.
+   */
+  object Annotations {
+
+    /**
+     * Ignores any None values in the case class
+     *
+     * {{{
+     * @IgnoreNone
+     * case class Person(name: String, nickName: Option[String])
+     *
+     * Person(name = "John", nickName: None) // becomes the equivalent of: { name: "John" }
+     * }}}
+     */
+    @meta.param
+    case class IgnoreNone() extends StaticAnnotation
```

Comment:
Well that is a valid case class. 

We could make the macro fail in this instance but I'm not sure of the need to check annotations at the field level, with this commit.